### PR TITLE
Input data structure

### PR DESCRIPTION
**Pull request by Doralitze**
This PR contains the basis for input desk processing.

### CODE REVIEW

File: src/io/iomanager.cpp [hunk 1]
```diff
@@ -10,30 +10,27 @@
...
 #include "proto_src/RealTimeControl.pb.h"
 #include "proto_src/UniverseControl.pb.h"
+COMPILER_RESTORE("-Wuseless-cast")
 #include "google/protobuf/io/zero_copy_stream.h"
 
 #include "io/universe_sender.hpp"
 #include "dmx/ftdi_universe.hpp"
 #include "xml/show_files.hpp"
 
+#include <iostream>
```
Comment (LRahlff) [inline src/io/iomanager.cpp:32]: do we need an iostream? is that usefull for the connection, or was it only for debugging (with cout)?

Comment (Doralitze) [reply src/io/iomanager.cpp:32]: I'll remove it quickly (debugging artefact)

File: src/io/iomanager.cpp [hunk 2]
```diff
@@ -510,6 +526,66 @@ void IOManager::parse_message_cb(uint32_t msg_type, client_handler& client){
             error_message += "IOManager Parse Message: Used MsgType_INT_MAX_SENTINEL_DO_NOT_USE_ as Msg Type. ";
             break;
         }
+        case ::missiondmx::fish::ipcmessages::MSGT_REMOVE_FADER_BANK_SET:
+            try {
+                missiondmx::fish::ipcmessages::remove_fader_bank_set msg;
+                if(!msg.ParseFromZeroCopyStream(buffer)) {
+                    error_message += "Failed to decode MSGT_REMOVE_FADER_BANK_SET message.";
+                }
+                if(control_desk_handle) {
```
Comment (LRahlff) [inline src/io/iomanager.cpp:535]: I thought i had (only) looked at iomanager, but here I dont understand your success/failure logic (my my code above is not nice as well) , but here you go ahead, also when  parsing fails...
and you dont return, so you get the error message for unknown msg types (with that message length)

Comment (Doralitze) [reply src/io/iomanager.cpp:535]: Line 558 features a break instruction, hence the error handling logic is correct. But the missing return is actually a bug and soon to be fixed.

Comment (Doralitze) [reply src/io/iomanager.cpp:535]: Fixed with commit 1b70296.

**Pull request merged**